src/GraphQL/GraphQLBuilderExtensions.cs

```diff
@@ -0,0 +1,827 @@
+#nullable enable
+
+using System;
+using System.Linq;
+using System.Reflection;
+using GraphQL.Caching;
+using GraphQL.DI;
+using GraphQL.Execution;
+using GraphQL.Instrumentation;
+using GraphQL.Types;
+using GraphQL.Types.Relay;
+using GraphQL.Utilities;
+using GraphQL.Validation;
+using GraphQL.Validation.Complexity;
+
+namespace GraphQL
+{
+    /// <summary>
+    /// Provides extension methods to configure GraphQL.NET services within a dependency injection framework.
+    /// </summary>
+    public static class GraphQLBuilderExtensions
+    {
+        #region - Additional overloads for Register, TryRegister, ConfigureDefaults and Configure -
+        /// <inheritdoc cref="Register{TService}(IGraphQLBuilder, Func{IServiceProvider, TService}, ServiceLifetime)"/>
+        public static IGraphQLBuilder Register<TService>(this IGraphQLBuilder graphQLBuilder, ServiceLifetime serviceLifetime)
```

Comment:
I would rename `graphQLBuilder` arg to `builder`.

Reply:
API change - not sure if that's considered breaking or not.  I'll just put it in v5.

Reply:
Not breaking. Approval file will change, yes, but it's not a breaking change.

Reply:
#2793

Reply:
#2793